### PR TITLE
Add union properrty option for Json merge

### PR DESCRIPTION
**Pull request by live1206**
For some properties, we need `Union` operation instead of `Replace`

https://dev.azure.com/ceapex/Engineering/_workitems/edit/157527

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5403)

### CODE REVIEW

File: src/docfx/build/page/BuildPage.cs [hunk 1]
```diff
@@ -93,6 +94,7 @@ public static async Task<List<Error>> Build(Context context, Document file)
             else
             {
                 JsonUtility.Merge(
+                    Array.Empty<string>(),
```
Comment (yufeih) [inline src/docfx/build/page/BuildPage.cs]: revert

File: src/docfx/config/ConfigLoader.cs [hunk 2]
```diff
@@ -67,7 +69,7 @@ where glob(configPath)
 
             // Preload
             var preloadConfigObject = new JObject();
-            JsonUtility.Merge(preloadConfigObject, envConfig, globalConfig, opsConfig, docfxConfig, cliConfig);
+            JsonUtility.Merge(unionProperties, preloadConfigObject, envConfig, globalConfig, opsConfig, docfxConfig, cliConfig);
```
Comment (yufeih) [inline src/docfx/config/ConfigLoader.cs]:
```suggestion
            JsonUtility.Merge(unionProperties, preloadConfigObject, envConfig, globalConfig, opsConfig, docfxConfig, options?.StdinConfig, cliConfig);
```

Comment (live1206) [reply src/docfx/config/ConfigLoader.cs]: would prefer to put CLI config separately

File: src/docfx/lib/json/JsonUtility.cs [hunk 3]
```diff
@@ -206,18 +206,18 @@ public static (List<Error> errors, JToken value) Parse(TextReader json, FilePath
             }
         }
 
-        public static void Merge(JObject container, params JObject[] overwrites)
+        public static void Merge(string[] unionProperties, JObject container, params JObject[] overwrites)
```
[the remark this answers is not in the document]
Comment (live1206) [reply src/docfx/lib/json/JsonUtility.cs]: overload method created above

**Comment by live1206** [comment]
@mingwli docs-help-pr build time exceeded 30s, is it a valid number?

**Pull request merged**